From 9e6df3eacebfc6282e0f5aabbfd7656932f3785c Mon Sep 17 00:00:00 2001
From: Alexandre Torgue <alexandre.torgue@foss.st.com>
Date: Thu, 17 Feb 2022 12:31:09 +0100
Subject: [PATCH 0081/1141] arm64: dts: st: add the optee firmware node on
 stm32mp25

Define the OP-TEE firmware node in the STM32MP2 device tree.

Signed-off-by: Alexandre Torgue <alexandre.torgue@foss.st.com>
Change-Id: If193a27b2b9b44ee4e1d6400ec3348d40dbdd8e0
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/240285
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
Reviewed-by: Lionel DEBIEVE <lionel.debieve@foss.st.com>
Reviewed-by: Etienne CARRIERE <etienne.carriere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/301078
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
Reviewed-by: Gabriel FERNANDEZ <gabriel.fernandez@foss.st.com>
Domain-Review: Patrick DELAUNAY <patrick.delaunay@foss.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 58b043f0ae71..12238f35bedc 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -56,6 +56,11 @@ ck_icn_ls_mcu: ck-icn-ls-mcu {
 	};
 
 	firmware {
+		optee {
+			compatible = "linaro,optee-tz";
+			method = "smc";
+		};
+
 		scmi: scmi {
 			compatible = "linaro,scmi-optee";
 			#address-cells = <1>;
-- 
2.39.2

